### PR TITLE
Drop support for Python 2.6.

### LINKED ISSUE #1607: Call for comments: Python 2.6 support will be dropped August 1, 2016
Anyone who thinks they still need Python 2.6, please speak up here. (Sorry there isn't a better channel.)

Relates to #995 

/cc @tseaver @jgeewax

### PR DESCRIPTION
**Pull request by tseaver**
Closes #1607.

### CODE REVIEW

**Comment by dhermes** [comment]
I'm `grep`-ing around for some other things worth cleaning up:

```
CONTRIBUTING.rst:- The feature must work fully on the following CPython versions: 2.6,
CONTRIBUTING.rst:-  `Python 2.6`_
CONTRIBUTING.rst:.. _Python 2.6: https://docs.python.org/2.6/
CONTRIBUTING.rst:We may `drop 2.6`_ as a supported version as well since Python 2.6 is no
CONTRIBUTING.rst:.. _drop 2.6: https://github.com/GoogleCloudPlatform/gcloud-python/issues/995
gcloud/_helpers.py:363:    if sys.version_info[:2] < (2, 7):  # pragma: NO COVER Python 2.6
gcloud/exceptions.py:40:        # suppress deprecation warning under 2.6.x
gcloud/error_reporting/client.py:174:            # once py26 support is dropped this can use dict comprehension
```

**Comment by tseaver** [comment]
@dhermes Should be addressed in subsequent commits.

**Comment by tseaver** [comment]
Amended to fix the lint issue.

File: gcloud/exceptions.py [hunk 1]
```diff
@@ -36,8 +36,7 @@ class GCloudError(Exception):
     """
 
     def __init__(self, message, errors=()):
-        super(GCloudError, self).__init__()
-        # suppress deprecation warning under 2.6.x
+        super(GCloudError, self).__init__(message)
         self.message = message
```
Comment (dhermes) [inline gcloud/exceptions.py]: You don't need to set `message` any more

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -81,8 +81,8 @@ In order to add a feature to ``gcloud-python``:
 - The feature must be documented in both the API and narrative
   documentation (in ``docs/``).
 
-- The feature must work fully on the following CPython versions: 2.6,
-  and 2.7 on both UNIX and Windows.
+- The feature must work fully on the following CPython versions:  2.7
+  and 3.4 on both UNIX and Windows.
```
Comment (dhermes) [inline CONTRIBUTING.rst]: We need to add 3.5 too

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -378,7 +376,7 @@ and lack of continuous integration `support`_.
 .. _decreased usage: https://caremad.io/2013/10/a-look-at-pypi-downloads/
 .. _support: http://blog.travis-ci.com/2013-11-18-upcoming-build-environment-updates/
 
-We may `drop 2.6`_ as a supported version as well since Python 2.6 is no
+We haved `dropped 2.6`_ as a supported version as well since Python 2.6 is no
```
Comment (dhermes) [inline CONTRIBUTING.rst]: s/haved/have

**Comment by dhermes** [comment]
Only question left: do we want to add a warning on import?

File: gcloud/exceptions.py (hunk 1 above)
Comment (tseaver) [inline gcloud/exceptions.py]: I tried that, but it made tests fail on Python 3. :(

```python
======================================================================
ERROR: test__request_not_200 (gcloud.datastore.test_connection.TestConnection)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/datastore/test_connection.py", line 170, in test__request_not_200
    self.assertEqual(str(e.exception), expected_message)
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/exceptions.py", line 43, in __str__
    return '%d %s' % (self.code, self.message)
AttributeError: 'BadRequest' object has no attribute 'message'

======================================================================
ERROR: test_ctor_defaults (gcloud.test_exceptions.Test_GCloudError)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/test_exceptions.py", line 30, in test_ctor_defaults
    self.assertEqual(str(e), '600 Testing')
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/exceptions.py", line 43, in __str__
    return '%d %s' % (self.code, self.message)
AttributeError: 'GCloudError' object has no attribute 'message'

======================================================================
ERROR: test_ctor_explicit (gcloud.test_exceptions.Test_GCloudError)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/test_exceptions.py", line 44, in test_ctor_explicit
    self.assertEqual(str(e), '600 Testing')
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/exceptions.py", line 43, in __str__
    return '%d %s' % (self.code, self.message)
AttributeError: 'GCloudError' object has no attribute 'message'

======================================================================
ERROR: test_hit_w_content_as_str (gcloud.test_exceptions.Test_make_exception)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/test_exceptions.py", line 62, in test_hit_w_content_as_str
    self.assertEqual(exception.message, 'Not Found')
AttributeError: 'NotFound' object has no attribute 'message'

======================================================================
ERROR: test_html_when_json_expected (gcloud.test_exceptions.Test_make_exception)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/test_exceptions.py", line 87, in test_html_when_json_expected
    self.assertEqual(exception.message, content)
AttributeError: 'NotFound' object has no attribute 'message'

======================================================================
ERROR: test_miss_w_content_as_dict (gcloud.test_exceptions.Test_make_exception)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tseaver/projects/agendaless/Google/src/gcloud-python/gcloud/test_exceptions.py", line 78, in test_miss_w_content_as_dict
    self.assertEqual(exception.message, 'Unknown Error')
AttributeError: 'GCloudError' object has no attribute 'message'

----------------------------------------------------------------------
```

Comment (dhermes) [inline gcloud/exceptions.py]: Oh. What's the Python 2.7/3+ accepted attribute name?

File: CONTRIBUTING.rst (hunk 3 above)
Comment (tseaver) [inline CONTRIBUTING.rst]: 28c6fdd

File: CONTRIBUTING.rst (hunk 2 above)
Comment (tseaver) [inline CONTRIBUTING.rst]: 4f5a55e

**Comment by tseaver** [comment]
> Do we want to add a warning on import? 

I think not on import:  *maybe* on installation.  @daspecster, @jgeewax WDYT?

File: gcloud/exceptions.py (hunk 1 above)
Comment (tseaver) [inline gcloud/exceptions.py]: We'd have to use `args[0]`.  I vote to leave `message` as our own custom attribute.

Comment (dhermes) [inline gcloud/exceptions.py]: OK. SGTM

**Comment by dhermes** [comment]
OK let's just merge (LGTM) and file an issue to follow up / decide if we should warn on either import or installation.

**Comment by tseaver** [comment]
See also #2056.

**Pull request merged**